base/shared/src/main/scala/scalaz/tc/foldable.scala

```diff
@@ -6,13 +6,24 @@ import scala.List
 import scala.language.experimental.macros
 
 trait FoldableClass[F[_]] {
+  /**
+  Applies a binary operator to a start value and all elements of this sequence, going left to right.
+  Note: this will not terminate for infinite-sized collections. For Example, Stream.
+  */
   def foldLeft[A, B](fa: F[A], z: B)(f: (B, A) => B): B
-
+  /**
+  Applies a binary operator to all elements of this list and a start value, going right to left.
+  */
   def foldRight[A, B](fa: F[A], z: => B)(f: (A, => B) => B): B
-
+  /**
+  It traverse the collection type A and does transform to B by using F function.
+  */
   def foldMap[A, B](fa: F[A])(f: A => B)(implicit B: Monoid[B]): B =
     foldRight(fa, B.mempty)((a, b) => B.mappend(f(a), b))
 
+  /**
+  It does addition by initializing with 0 with foldLeft and Identity value of particular monoid.
+  */
```

Comment:
This one is incorrect. `Monoid a => a` can be any monoid, not just addition with 0. It could be a function composition, string appends with "", list append with [], multiplication with 1, modular arithmetic with 0, etc. This just defers to the monoid instance to choose how the fold will be executed.